app/models/spree/product_set.rb

```diff
@@ -89,10 +89,19 @@ def create_variant(product, variant_attributes)
     on_demand = variant_attributes.delete(:on_demand)
 
     variant = product.variants.create(variant_attributes)
-    return unless variant.valid?
 
-    variant.on_demand = on_demand if on_demand.present?
-    variant.on_hand = on_hand.to_i if on_hand.present?
+    begin
+      variant.on_demand = on_demand if on_demand.present?
+      variant.on_hand = on_hand.to_i if on_hand.present?
+    rescue StandardError => error
+      Bugsnag.notify(error) do |report|
+        report.add_tab(:product, product.attributes)
+        report.add_tab(:product_error, product.errors.first) unless product.valid?
+        report.add_tab(:variant_attributes, variant_attributes)
+        report.add_tab(:variant, variant.attributes)
+        report.add_tab(:variant_error, variant.errors.first) unless variant.valid?
+      end
+    end
```

Comment:
@luisramos0 Do we really want this? This would make saving appear successful, and the user might not realize that their data was not saved.

I think what we really need is feedback in the front-end pointing the user to the problem with their data, but of course that would take more time.

Reply:
yes, agree. I dont think I can go for that right now: add user feedback in the frontend.

Do you think it's better to raise the original error and let the user know there's something wrong? I can do that.

Reply:
ok, [new commit](https://github.com/openfoodfoundation/openfoodnetwork/pull/4159/commits/249a3c4e182eed65e7158251a34c03a1ced5fcfe) to make the exception fly out to the user, we keep the bugsnag info.

Reply:
I have updated PR description as well.